--- qtbase-opensource-src-5.2.1/src/plugins/platforms/eglfs/eglfs.pri.orig	2014-05-27 18:45:06.515816122 +0800
+++ qtbase-opensource-src-5.2.1/src/plugins/platforms/eglfs/eglfs.pri	2014-05-27 18:45:19.871816373 +0800
@@ -3,12 +3,12 @@ QT += core-private gui-private platforms
 #DEFINES += QEGL_EXTRA_DEBUG
 
 # Avoid X11 header collision
-DEFINES += MESA_EGL_NO_X11_HEADERS
+# DEFINES += MESA_EGL_NO_X11_HEADERS
 
 # Uncomment these to enable the X hooks, allowing to test the platform
 # plugin in a regular X11 environment (as long as EGL is available).
 # EGLFS_PLATFORM_HOOKS_SOURCES += qeglfshooks_x11.cpp
-# LIBS += -lX11 -lX11-xcb -lxcb
+LIBS += -lX11 -lX11-xcb -lxcb
 
 SOURCES +=  $$PWD/qeglfsintegration.cpp \
             $$PWD/qeglfswindow.cpp \
