diff -rNu swupdate-2018.11-Orig/corelib/stream_interface.c swupdate-2018.11/corelib/stream_interface.c
--- swupdate-2018.11-Orig/corelib/stream_interface.c	2019-03-21 17:40:01.077317581 +0800
+++ swupdate-2018.11/corelib/stream_interface.c	2019-03-21 17:40:19.045318110 +0800
@@ -413,9 +413,10 @@
 			if (ret == 0) {
 				close(inst.fd);
 				inst.last_install = SUCCESS;
-				bootloader_env_set("upgrade_step", "3");
+//				bootloader_env_set("upgrade_step", "3");
 				notify(SUCCESS, RECOVERY_NO_ERROR, INFOLEVEL, "SWUPDATE download successful !, reboot...");
 				umount("/data");
+				set_recovery();
 				system("reboot");
 				sleep(5);
 			} else {
