diff --git a/src/parsing.cpp b/src/parsing.cpp
index bb5ff69..1ebf8ce 100644
--- a/src/parsing.cpp
+++ b/src/parsing.cpp
@@ -191,7 +191,8 @@ bool parse_args(std::shared_ptr<t_config> &config, int argc, char **argv) {
               "input] [-u username] [-p password] [-f framerate] [-s "
               "'width'x'height'] [-d] [-t] [-h] [-g gst-pipeline]\n",
               argv[0]);
-      return true;
+      exit(0);
+      break;
     case '?':
       if (optopt == 'r' || optopt == 'l' || optopt == 'p' || optopt == 'u' ||
           optopt == 'i' || optopt == 'a' || optopt == 'b' || optopt == 'f' ||
diff --git a/src/server.cpp b/src/server.cpp
index a4f82d3..74eaea9 100644
--- a/src/server.cpp
+++ b/src/server.cpp
@@ -65,8 +65,10 @@ bool server_init(t_server *serv) {
                            serv->config->username.c_str(), NULL);
 
   if (serv->config->digest) {
-    gst_rtsp_auth_set_supported_methods(serv->auth, GST_RTSP_AUTH_DIGEST);
-    gst_rtsp_auth_add_digest(serv->auth, serv->config->username.c_str(), serv->config->password.c_str(), serv->token);
+    //gst_rtsp_auth_set_supported_methods(serv->auth, GST_RTSP_AUTH_DIGEST);
+    //gst_rtsp_auth_add_digest(serv->auth, serv->config->username.c_str(), serv->config->password.c_str(), serv->token);
+    g_print("digest is not supported in this version\n");
+    return false;
   } else {
     serv->basic = gst_rtsp_auth_make_basic(serv->config->username.c_str(),
                                            serv->config->password.c_str());
