diff --git a/src/pipeline.cpp b/src/pipeline.cpp
index f8636aa..db471e5 100644
--- a/src/pipeline.cpp
+++ b/src/pipeline.cpp
@@ -49,8 +49,9 @@ std::string encode(std::shared_ptr<t_config> &config) {
   launchCmd += " ! videorate ! video/x-raw,framerate=";
   launchCmd += config->framerate;
   launchCmd += "/1";
-  launchCmd += " ! capsfilter ! queue"
-               " ! x264enc speed-preset=superfast";
+  launchCmd += " ! amlvenc";
+  launchCmd += " framerate=";
+  launchCmd += config->framerate;
 
   return launchCmd;
 }
