diff --git a/include/server.h b/include/server.h
index c46a1a9..cbd4fc5 100644
--- a/include/server.h
+++ b/include/server.h
@@ -63,6 +63,7 @@ typedef struct s_config {
   bool face_detection;
   // use x265
   bool use_x265;
+  bool disable_videorate;
 } t_config;
 
 typedef struct s_server {
diff --git a/src/parsing.cpp b/src/parsing.cpp
index fb37bbe..b4a3298 100644
--- a/src/parsing.cpp
+++ b/src/parsing.cpp
@@ -128,6 +128,13 @@ void parse_env(std::shared_ptr<t_config> &config) {
       config->use_x265 = false;
     }
   }
+  if ( const char *disable_videorate = std::getenv("RTSP_DISABLE_VIDEORATE")) {
+    if (strcmp(disable_videorate, "true") == 0) {
+      config->disable_videorate = true;
+    } else {
+      config->disable_videorate = false;
+    }
+  }
 }
 
 // Overwrite default parameters via cmd line
diff --git a/src/pipeline.cpp b/src/pipeline.cpp
index cf82866..557e66d 100644
--- a/src/pipeline.cpp
+++ b/src/pipeline.cpp
@@ -56,9 +56,11 @@ std::string encode(std::shared_ptr<t_config> &config) {
   if (config->face_detection) {
     launchCmd += " ! amlyoloface";
   }
-  launchCmd += " ! videorate ! video/x-raw,framerate=";
-  launchCmd += config->framerate;
-  launchCmd += "/1";
+  if (!config->disable_videorate) {
+    launchCmd += " ! videorate ! video/x-raw,framerate=";
+    launchCmd += config->framerate;
+    launchCmd += "/1";
+  }
   if (config->use_x265) {
     launchCmd += " ! amlx265enc";
   } else {
