Index: weston-1.10.0/clients/desktop-shell.c
===================================================================
--- weston-1.10.0.orig/clients/desktop-shell.c
+++ weston-1.10.0/clients/desktop-shell.c
@@ -1015,7 +1015,7 @@ background_create(struct desktop *deskto
 	background->widget = window_add_widget(background->window, background);
 	window_set_user_data(background->window, background);
 	widget_set_redraw_handler(background->widget, background_draw);
-	widget_set_transparent(background->widget, 0);
+	widget_set_transparent(background->widget, 1);
 	window_set_preferred_format(background->window,
 				    WINDOW_PREFERRED_FORMAT_RGB565);
 
