--- xdriver_xf86-video-mali-master/configure.ac.old	2014-04-30 12:14:13.766319208 +0800
+++ xdriver_xf86-video-mali-master/configure.ac	2014-04-30 14:08:56.262448329 +0800
@@ -56,7 +56,7 @@ XORG_DRIVER_CHECK_EXT(RENDER, renderprot
 XORG_DRIVER_CHECK_EXT(XV, videoproto)
 
 # Checks for pkg-config packages
-PKG_CHECK_MODULES(XORG, [xorg-server >= 1.0.99.901 xproto fontsproto $REQUIRED_MODULES])
+PKG_CHECK_MODULES(XORG, [xorg-server >= 1.0.99.901 xproto fontsproto libdrm $REQUIRED_MODULES])
 sdkdir=$(pkg-config --variable=sdkdir xorg-server)
 
 save_CFLAGS="$CFLAGS"
@@ -75,6 +75,28 @@ if test "x$PCIACCESS" = xyes; then
 fi
 
 # Checks for libraries.
+save_CFLAGS="$CFLAGS"
+CFLAGS="$CFLAGS -pthread" 
+
+AC_CHECK_HEADER([ump/ump.h], [have_libump=yes], [have_libump=no])
+
+if test "x$have_libump" = xyes; then
+    AC_CHECK_LIB([UMP], [ump_open], [have_libump=yes], [have_libump=no])
+fi
+
+if test "x$have_libump" = xyes; then
+    LIBS="$LIBS -lUMP"
+    AC_DEFINE(HAVE_LIBUMP,[1],[libUMP library])
+    AC_CHECK_LIB([UMP], [ump_cache_operations_control],
+                 [AC_DEFINE(HAVE_LIBUMP_CACHE_CONTROL,[1],[UMP cache control])])
+fi
+
+AM_CONDITIONAL([HAVE_LIBUMP], [test x$have_libump = xyes])
+
+# revert back to the original CFLAGS if not messing with libUMP
+if test "x$have_libump" = xno; then
+    CFLAGS="$save_CFLAGS"
+fi
 
 # Checks for header files.
 AC_HEADER_STDC
