--- uboot-next/board/amlogic/configs/m8_k200_v1.h.orig	2014-04-16 14:46:33.403797826 +0800
+++ uboot-next/board/amlogic/configs/m8_k200_v1.h	2014-04-16 14:53:37.791805788 +0800
@@ -255,7 +255,7 @@
 	"sdcburncfg=aml_sdc_burn.ini\0"\
 	"normalstart=1000000\0" \
 	"normalsize=400000\0" \
-	"upgrade_step=0\0" \
+	"upgrade_step=2\0" \
 	"firstboot=1\0" \
 	"store=0\0"\
 	"preboot="\
@@ -293,6 +293,9 @@
    	"storeargs="\
         "setenv bootargs ${bootargs} vdaccfg=${vdac_config} logo=osd1,loaded,${fb_addr},${outputmode},full hdmimode=${hdmimode} cvbsmode=${cvbsmode} androidboot.firstboot=${firstboot} hdmitx=${cecconfig}\0"\
     \
+    "bootsdargs="\
+        "root=/dev/mmcblk0p1 rw rootfstype=ext2 rootwait init=/init console=ttyS0,115200n8 no_console_suspend logo=osd1,loaded,${fb_addr},${outputmode},full hdmimode=${hdmimode} cvbsmode=${cvbsmode} hdmitx=${cecconfig}\0"\
+    \
 	"switch_bootmode="\
         "if test ${reboot_mode} = factory_reset; then "\
 			"run recovery;"\
@@ -317,9 +320,14 @@
             "setenv bootargs ${bootargs} androidboot.serialno=${usid};"\
         "fi;"\
         "imgread kernel boot ${loadaddr};"\
-        "bootm;"\
-        "run recovery\0" \
+        "bootm\0"\
     \
+     "bootsdcard="\
+        "echo Booting ...;"\
+         "setenv bootargs ${bootsdargs}; "\
+         "mmcinfo;"\
+         "ext2load mmc 0 ${loadaddr} boot.img;"\
+         "bootm\0" \
 	"recovery="\
         "echo enter recovery;"\
         "if mmcinfo; then "\
