diff --git a/samples/opengles_20/rotozoom/RotoZoom.cpp b/samples/opengles_20/rotozoom/RotoZoom.cpp
index 2dd23af..a46d295 100644
--- a/samples/opengles_20/rotozoom/RotoZoom.cpp
+++ b/samples/opengles_20/rotozoom/RotoZoom.cpp
@@ -27,8 +27,8 @@
 #include "Mathematics.h"
 #include "EGLRuntime.h"
 
-#define WINDOW_W 480
-#define WINDOW_H 800
+#define WINDOW_W 800
+#define WINDOW_H 480
 
 using std::string;
 using namespace MaliSDK;
