Index: aml_nand-amlogic-nand/amlnf/Makefile
===================================================================
--- aml_nand-amlogic-nand.orig/amlnf/Makefile	2014-09-15 17:13:26.000000000 +0800
+++ aml_nand-amlogic-nand/amlnf/Makefile	2014-09-23 11:39:08.974971968 +0800
@@ -35,7 +35,7 @@
 aml_nftl_dev-objs += block/aml_nftl_block.o
 aml_nftl_dev-objs += block/aml_nftl_init.o
 aml_nftl_dev-objs += block/aml_nftl_hw_interface.o
-aml_nftl_dev-objs += ../../../../../../../../../hardware/amlogic/nand/amlnf/nftl/aml_nftl_core_20140911.z
+aml_nftl_dev-objs += nftl/aml_nftl_core_20140911.z
 
 #aml_nftl_dev-objs += nftl/aml_nftl_phy.o       
 #aml_nftl_dev-objs  += nftl/aml_nftl_list.o      
