--- aml_nand-amlogic-nand/amlnf/Makefile.orig	2014-04-02 14:55:14.717114758 +0800
+++ aml_nand-amlogic-nand/amlnf/Makefile	2014-04-02 14:55:28.781115022 +0800
@@ -14,7 +14,7 @@ aml_nftl_dev-objs =
 aml_nftl_dev-objs += block/aml_nftl_block.o
 aml_nftl_dev-objs += block/aml_nftl_init.o
 aml_nftl_dev-objs += block/aml_nftl_hw_interface.o
-aml_nftl_dev-objs += ../../../../../../../../../hardware/amlogic/nand/amlnf/nftl/aml_nftl_core_20140324.z
+aml_nftl_dev-objs += nftl/aml_nftl_core_20140324.z
 
 #aml_nftl_dev-objs += nftl/aml_nftl_phy.o       
 #aml_nftl_dev-objs  += nftl/aml_nftl_list.o      
--- aml_nand-amlogic-nand/aml_nftl_new/Makefile.orig	2014-04-02 14:54:40.789114121 +0800
+++ aml_nand-amlogic-nand/aml_nftl_new/Makefile	2014-04-02 14:54:52.933114349 +0800
@@ -4,6 +4,6 @@ aml_nftl_dev-objs += block/aml_nftl_bloc
 aml_nftl_dev-objs += block/aml_nftl_init.o
 aml_nftl_dev-objs += block/aml_nftl_hw_interface.o
 
-aml_nftl_dev-objs += ../../../../../../../../../hardware/amlogic/nand/aml_nftl_new/nftl/aml_nftl_core_20131226.z
+aml_nftl_dev-objs += nftl/aml_nftl_core_20131226.z
 
 obj-$(CONFIG_AML_NFTL_NEW)	+= aml_nftl_dev.o
