--- brcmap6xxx-ap6xxx/broadcm_40181/Kconfig.orig	2014-01-13 18:18:50.926398265 +0800
+++ brcmap6xxx-ap6xxx/broadcm_40181/Kconfig	2014-01-13 18:19:09.774398625 +0800
@@ -10,21 +10,21 @@ config SDIO_DHD_CDC_WIFI_40181_MODULE
 config BCMDHD_FW_PATH
 	depends on SDIO_DHD_CDC_WIFI_40181_MODULE
 	string "Firmware path"
-	default "/system/etc/wifi/fw_bcmdhd.bin"
+	default "/etc/wifi/fw_bcmdhd.bin"
 	---help---
 	  Path to the firmware file.
 
 config BCMDHD_NVRAM_PATH
 	depends on SDIO_DHD_CDC_WIFI_40181_MODULE
 	string "NVRAM path"
-	default "/system/etc/wifi/nvram.txt"
+	default "/etc/wifi/nvram.txt"
 	---help---
 	  Path to the calibration file.
 
 config BCMDHD_CONFIG_PATH
 	depends on SDIO_DHD_CDC_WIFI_40181_MODULE
 	string "Config path"
-	default "/system/etc/firmware/config.txt"
+	default "/etc/firmware/config.txt"
 	---help---
 	  Path to the driver configuration file.

--- brcmap6xxx-ap6xxx/broadcm_40181/Makefile.orig	2014-01-23 15:55:30.086446084 +0800
+++ brcmap6xxx-ap6xxx/broadcm_40181/Makefile	2014-01-23 15:56:59.266447768 +0800
@@ -13,7 +13,7 @@ DHDCFLAGS = -Wall -Wstrict-prototypes -D
 	-DDHD_DONOT_FORWARD_BCMEVENT_AS_NETWORK_PKT -DAMPDU_HOSTREORDER       \
 	-DCUSTOM_SDIO_F2_BLKSIZE=128 -DWL_SDO -DWLTDLS                        \
 	-DWL_SUPPORT_BACKPORTED_KPATCHES                                      \
-	-I../hardware/wifi/broadcom/drivers/ap6xxx/broadcm_40181/ -I../hardware/wifi/broadcom/drivers/ap6xxx/broadcm_40181/include
+	-Idrivers/amlogic/wifi/broadcm_40181/ -Idrivers/amlogic/wifi/broadcm_40181/include
 
 DHDOFILES = aiutils.o bcmsdh_sdmmc_linux.o dhd_linux.o siutils.o bcmutils.o   \
 	dhd_linux_sched.o bcmwifi_channels.o dhd_sdio.o bcmevent.o dhd_bta.o hndpmu.o       \
