--- linux-amlogic-3.0/drivers/amlogic/uart/uart/am_uart.c.orig	2014-03-25 14:34:51.444281257 +0800
+++ linux-amlogic-3.0/drivers/amlogic/uart/uart/am_uart.c	2014-03-25 14:35:39.484282489 +0800
@@ -620,9 +620,11 @@ am_uart_set_termios(struct uart_port *po
 #endif
 
     baud = tty_termios_baud_rate(termios);
-    change_speed(info, baud);
+    if (baud) {
+        change_speed(info, baud);
 
-    uart_update_timeout(port, termios->c_cflag, baud);
+        uart_update_timeout(port, termios->c_cflag, baud);
+    }
 
     return;
 }
--- linux-amlogic-3.0/drivers/amlogic/uart/uart/meson_uart.c.orig	2014-03-25 14:35:01.428281606 +0800
+++ linux-amlogic-3.0/drivers/amlogic/uart/uart/meson_uart.c	2014-03-25 14:35:48.628282601 +0800
@@ -631,9 +631,11 @@ meson_uart_set_termios(struct uart_port
     //    return;
 
     baud = tty_termios_baud_rate(termios);
-    change_speed(info, baud);
+    if (baud) {
+        change_speed(info, baud);
 
-    uart_update_timeout(port, termios->c_cflag, baud);
+        uart_update_timeout(port, termios->c_cflag, baud);
+    }
 
     return;
 }
