Index: uboot-next-2015-dev/include/configs/gxb_p201_v1.h
===================================================================
--- uboot-next-2015-dev.orig/include/configs/gxb_p201_v1.h	2015-09-07 17:32:12.000000000 +0800
+++ uboot-next-2015-dev/include/configs/gxb_p201_v1.h	2015-09-07 18:24:18.255812042 +0800
@@ -99,7 +99,15 @@
             "fi;"\
             "\0" \
         "storeargs="\
-            "setenv bootargs ${initargs} logo=${display_layer},loaded,${fb_addr},${outputmode},hdmimode=${hdmimode} cvbsmode=${cvbsmode} hdmitx=${cecconfig} androidboot.firstboot=${firstboot}; "\
+            "setenv bootargs root=/dev/system rootfstype=ext4 ${initargs} logo=${display_layer},loaded,${fb_addr},${outputmode},hdmimode=${hdmimode} cvbsmode=${cvbsmode} hdmitx=${hdmimode} androidboot.firstboot=${firstboot}; "\
+            "run cmdline_keys;"\
+            "\0"\
+        "sdcardargs="\
+            "setenv bootargs root=/dev/mmcblk0p1 rootfstype=ext2 ${initargs} logo=${display_layer},loaded,${fb_addr},${outputmode},hdmimode=${hdmimode} cvbsmode=${cvbsmode} hdmitx=${hdmimode} androidboot.firstboot=${firstboot}; "\
+            "run cmdline_keys;"\
+            "\0"\
+        "bootupdateargs="\
+            "setenv bootargs root=/dev/mmcblk0p1 rootfstype=vfat ${initargs} logo=${display_layer},loaded,${fb_addr},${outputmode},hdmimode=${hdmimode} cvbsmode=${cvbsmode} hdmitx=${hdmimode} androidboot.firstboot=${firstboot} firmware=rootfs.tar.gz; "\
             "run cmdline_keys;"\
             "\0"\
         "switch_bootmode="\
@@ -116,6 +124,12 @@
             "if imgread kernel boot ${loadaddr}; then store dtb read $dtb_mem_addr; bootm ${loadaddr}; fi;"\
             "run update;"\
             "\0"\
+        "bootsdcard="\
+            "run sdcardargs; mmcinfo; ext2load mmc 0 ${loadaddr} boot.img; bootm ${loadaddr};"\
+            "\0"\
+        "bootupdate="\
+            "run bootupdateargs; mmcinfo; fatload mmc 0 ${loadaddr} boot.img; bootm ${loadaddr};"\
+            "\0"\
         "factory_reset_poweroff_protect="\
             "echo wipe_data=${wipe_data}; echo wipe_cache=${wipe_cache};"\
             "if test ${wipe_data} = failed; then "\
@@ -322,7 +336,8 @@
 #define CONFIG_FS_FAT 1
 #define CONFIG_FS_EXT4 1
 #define CONFIG_LZO 1
-
+#define CONFIG_CMD_EXT2 1
+#define CONFIG_CMD_EXT4 1
 /* Cache Definitions */
 //#define CONFIG_SYS_DCACHE_OFF
 //#define CONFIG_SYS_ICACHE_OFF
