Index: uboot-2015-dev/bl33/Makefile
===================================================================
--- uboot-2015-dev.orig/bl33/Makefile	2017-07-20 15:29:13.958496982 +0800
+++ uboot-2015-dev/bl33/Makefile	2017-07-20 15:47:27.511958613 +0800
@@ -247,6 +247,7 @@ CROSS_COMPILE ?=
 endif
 
 export CROSS_COMPILE=aarch64-none-elf-
+export CROSS_COMPILE_T32=arm-none-eabi-
 
 KCONFIG_CONFIG	?= .config
 export KCONFIG_CONFIG
