diff -Nur uboot-2019-orig/fip/variables.sh uboot-2019/fip/variables.sh
--- uboot-2019-orig/fip/variables.sh	2019-10-31 14:21:56.547947492 +0800
+++ uboot-2019/fip/variables.sh	2019-10-31 14:33:06.712956141 +0800
@@ -29,8 +29,8 @@
 declare BL33_PATH1="bl33/v2015"
 declare BL33_PATH2="bl33/v2019"
 
-declare BL33_TOOLCHAIN1="aarch64-none-elf-"
-declare BL33_TOOLCHAIN2="/opt/gcc-linaro-7.3.1-2018.05-i686_aarch64-elf/bin/aarch64-elf-"
+declare BL33_TOOLCHAIN1="${CROSS_COMPILE}"
+declare BL33_TOOLCHAIN2="${CROSS_COMPILE}"
 
 declare BL33_DEFCFG1="${UBOOT_FOLDER}/${UBOOT_VERSION1}/${DEFCFG_FOLDER}"
 declare BL33_DEFCFG2="${UBOOT_FOLDER}/${UBOOT_VERSION2}/${DEFCFG_FOLDER}"
@@ -44,4 +44,4 @@
 	export MANIFEST
 	export CUR_SOC
 	export BOARD_DIR
-}
\ No newline at end of file
+}
