diff -Naur icu-4.2.1.orig/source/config/mh-linux icu-4.2.1/source/config/mh-linux
--- icu-4.2.1.orig/source/config/mh-linux	2013-11-02 14:53:08.102359138 -0700
+++ icu-4.2.1/source/config/mh-linux	2013-11-02 14:50:03.994352895 -0700
@@ -32,7 +32,7 @@
 ## Shared object suffix
 SO = so
 ## Non-shared intermediate object suffix
-STATIC_O = ao
+STATIC_O = o
 
 ## Compilation rules
 %.$(STATIC_O): $(srcdir)/%.c
