diff --git a/arch/arm64/configs/odroidg12_defconfig b/arch/arm64/configs/odroidg12_defconfig
index 338c474..87a8d6c 100644
--- a/arch/arm64/configs/odroidg12_defconfig
+++ b/arch/arm64/configs/odroidg12_defconfig
@@ -2877,8 +2877,7 @@ CONFIG_ZD1211RW=m
 # CONFIG_ZD1211RW_DEBUG is not set
 CONFIG_88XXAU=m
 CONFIG_RTL8188EU=m
-CONFIG_ESP8089=m
-CONFIG_ESP8089_DEBUG_FS=y
+# CONFIG_ESP8089 is not set
 CONFIG_RTL8821CU=m
 # CONFIG_MAC80211_HWSIM is not set
 CONFIG_USB_NET_RNDIS_WLAN=m
diff --git a/drivers/net/wireless/rtl8812au/Makefile b/drivers/net/wireless/rtl8812au/Makefile
index 757145c..6828f14 100755
--- a/drivers/net/wireless/rtl8812au/Makefile
+++ b/drivers/net/wireless/rtl8812au/Makefile
@@ -5,7 +5,7 @@ EXTRA_CFLAGS += -Wno-unused-variable
 EXTRA_CFLAGS += -Wno-unused-label
 #EXTRA_CFLAGS += -Wno-unused-parameter
 EXTRA_CFLAGS += -Wno-unused-function
-EXTRA_CFLAGS += -Wimplicit-fallthrough=0
+#EXTRA_CFLAGS += -Wimplicit-fallthrough=0
 #EXTRA_CFLAGS += -Wno-parentheses-equality
 #EXTRA_CFLAGS += -Wno-pointer-bool-conversion
 EXTRA_CFLAGS += -Wno-unknown-pragmas
