--- linux-amlogic-3.10-bringup/drivers/gpu/drm/Kconfig.orig	2014-04-17 15:37:31.133476107 +0800
+++ linux-amlogic-3.10-bringup/drivers/gpu/drm/Kconfig	2014-04-17 15:37:49.333476449 +0800
@@ -199,6 +199,13 @@ config DRM_SAVAGE
 	  Choose this option if you have a Savage3D/4/SuperSavage/Pro/Twister
 	  chipset. If M is selected the module will be called savage.
 
+config DRM_MALI
+	tristate "arm mali cards"
+	depends on DRM
+	help
+	  Choose this option if you have a ARM GPU Mali.
+	  If M is selected the module will be called arm.
+
 source "drivers/gpu/drm/exynos/Kconfig"
 
 source "drivers/gpu/drm/vmwgfx/Kconfig"
