Index: linux-amlogic-3.10-bringup/drivers/amlogic/ethernet/am_net8218.c
===================================================================
--- linux-amlogic-3.10-bringup.orig/drivers/amlogic/ethernet/am_net8218.c	2014-09-04 14:10:43.000000000 +0800
+++ linux-amlogic-3.10-bringup/drivers/amlogic/ethernet/am_net8218.c	2014-09-12 17:06:58.033510745 +0800
@@ -902,7 +902,7 @@
 	char *endp;
 	int j;
 	ret = get_aml_key_kernel("mac", print_buff, 0);
-	extenal_api_key_set_version("nand3");
+	extenal_api_key_set_version("auto3");
 	printk("ret = %d\nprint_buff=%s\n", ret, print_buff);
 	if (ret >= 0) {
 		strcpy(ndev->dev_addr, print_buff);
