--- linux-amlogic-3.10-bringup/drivers/amlogic/Makefile.orig	2014-03-01 19:39:40.000000000 +0800
+++ linux-amlogic-3.10-bringup/drivers/amlogic/Makefile	2014-03-06 11:14:21.648208244 +0800
@@ -33,9 +33,9 @@ obj-y += cpufreq/
 obj-$(CONFIG_GPIO_AMLOGIC) += gpio/
 obj-y += pinctrl/
 
-ifneq ($(wildcard $(srctree)/../hardware/amlogic/nand),)
-	obj-$(CONFIG_AML_NEXT_GEN_NAND) += ../../../hardware/amlogic/nand/amlnf/
-	obj-$(CONFIG_AM_NAND) += ../../../hardware/amlogic/nand/nand/
+ifneq ($(wildcard $(srctree)/drivers/amlogic/nand),)
+	obj-$(CONFIG_AML_NEXT_GEN_NAND) += nand/amlnf/
+	obj-$(CONFIG_AM_NAND) += nand/nand/
 endif
 
 obj-y += ethernet/
@@ -68,8 +68,8 @@ obj-$(CONFIG_AML_RTC)+= rtc/
 obj-y += power/
 
 obj-y += tvin_v4l2/
-ifneq ($(wildcard $(srctree)/../hardware/tvin),)
-	obj-$(CONFIG_TVIN) += ../../../hardware/tvin/
+ifneq ($(wildcard $(srctree)/drivers/amlogic/tvin),)
+	obj-$(CONFIG_TVIN) += tvin/
 endif
 
 obj-y += thermal/
@@ -87,13 +87,13 @@ obj-$(CONFIG_AM_DVB) += dvb_tv/
 obj-$(CONFIG_AM_SMARTCARD) += smartcard/
 
 obj-$(CONFIG_AM_WIFI) += wifi/
-ifneq ($(wildcard $(srctree)/../hardware/amlogic/wifi),)
-	obj-$(CONFIG_SDIO_DHD_CDC_WIFI_40181_MODULE) += ../../../hardware/wifi/broadcom/drivers/ap6xxx/broadcm_40181/
-	obj-$(CONFIG_RTL8188EU) += ../../../hardware/wifi/realtek/drivers/8188eu/rtl8xxx_EU/
-	obj-$(CONFIG_RTL8188EU_MP) += ../../../hardware/wifi/realtek/drivers/8188eu/rtl8xxx_EU_MP/
-	obj-$(CONFIG_RTL8192CU) += ../../../hardware/wifi/realtek/drivers/8192cu/rtl8xxx_CU/
-	obj-$(CONFIG_RTL8192DU) += ../../../hardware/wifi/realtek/drivers/8192du/rtl8xxx_DU/
-	obj-$(CONFIG_RTL8192EU) += ../../../hardware/wifi/realtek/drivers/8192eu/rtl8192EU/
+ifneq ($(wildcard $(srctree)/drivers/amlogic/wifi),)
+	obj-$(CONFIG_SDIO_DHD_CDC_WIFI_40181_MODULE) += wifi/broadcm_40181/
+	obj-$(CONFIG_RTL8188EU) += wifi/rtl8xxx_EU/
+	obj-$(CONFIG_RTL8188EU_MP) += wifi/rtl8xxx_EU_MP/
+	obj-$(CONFIG_RTL8192CU) += wifi/rtl8xxx_CU/
+	obj-$(CONFIG_RTL8192DU) += wifi/rtl8xxx_DU/
+	obj-$(CONFIG_RTL8192EU) += wifi/rtl8192EU/
 endif
 obj-y += bluetooth/
 obj-$(CONFIG_MESON_NFC) += nfc/
@@ -104,9 +104,12 @@ obj-$(CONFIG_AMLOGIC_MIPI) += mipi/
 
 obj-$(CONFIG_AML_REG_DEBUG) += regd/
 
-ifneq ($(wildcard $(srctree)/../hardware/arm/gpu),)
-	obj-$(CONFIG_UMP) += ../../../hardware/arm/gpu/ump/
-	obj-$(CONFIG_MALI400) += ../../../hardware/arm/gpu/mali/
+ifneq ($(wildcard $(srctree)/drivers/amlogic/ump),)
+	obj-$(CONFIG_UMP) += ump/
+endif
+
+ifneq ($(wildcard $(srctree)/drivers/amlogic/mali),)
+	obj-$(CONFIG_MALI400) += mali/
 endif
 
 obj-$(CONFIG_AMLOGIC_ION) += ion_dev/
--- linux-amlogic-3.10-bringup/drivers/amlogic/Kconfig.orig	2014-03-01 19:39:40.000000000 +0800
+++ linux-amlogic-3.10-bringup/drivers/amlogic/Kconfig	2014-03-06 11:15:35.960209638 +0800
@@ -34,8 +34,8 @@ source "drivers/amlogic/usb/Kconfig"
 #source "drivers/amlogic/cardreader/Kconfig"
 source "drivers/amlogic/mmc/Kconfig"
 source "drivers/amlogic/spi_nor/Kconfig"
-isource "../hardware/amlogic/nand/amlnf/Kconfig"
-isource "../hardware/amlogic/nand/nand/Kconfig"
+isource "drivers/amlogic/nand/amlnf/Kconfig"
+isource "drivers/amlogic/nand/nand/Kconfig"
 
 #
 #	Network devices
@@ -43,12 +43,12 @@ isource "../hardware/amlogic/nand/nand/K
 comment "Network devices"
 source "drivers/amlogic/ethernet/Kconfig"
 source "drivers/amlogic/wifi/Kconfig"
-isource "../hardware/wifi/broadcom/drivers/ap6xxx/broadcm_40181/Kconfig"
-isource "../hardware/wifi/realtek/drivers/8188eu/rtl8xxx_EU/Kconfig"
-isource "../hardware/wifi/realtek/drivers/8188eu/rtl8xxx_EU_MP/Kconfig"
-isource "../hardware/wifi/realtek/drivers/8192cu/rtl8xxx_CU/Kconfig"
-isource "../hardware/wifi/realtek/drivers/8192du/rtl8xxx_DU/Kconfig"
-isource "../hardware/wifi/realtek/drivers/8192eu/rtl8192EU/Kconfig"
+isource "drivers/amlogic/wifi/broadcm_40181/Kconfig"
+isource "drivers/amlogic/wifi/rtl8xxx_EU/Kconfig"
+isource "drivers/amlogic/wifi/rtl8xxx_EU_MP/Kconfig"
+isource "drivers/amlogic/wifi/rtl8xxx_CU/Kconfig"
+isource "drivers/amlogic/wifi/rtl8xxx_DU/Kconfig"
+isource "drivers/amlogic/wifi/rtl8192EU/Kconfig"
 source "drivers/amlogic/bluetooth/Kconfig"
 source "drivers/amlogic/nfc/Kconfig"
 
@@ -67,7 +67,7 @@ source "drivers/amlogic/amports/Kconfig"
 source "drivers/amlogic/canvas/Kconfig"
 source "drivers/amlogic/display/Kconfig"
 source "drivers/amlogic/hdmi/Kconfig"
-isource "../hardware/tvin/Kconfig"
+isource "drivers/amlogic/tvin/Kconfig"
 source "drivers/amlogic/ppmgr/Kconfig"
 source "drivers/amlogic/camera/Kconfig"
 source "drivers/amlogic/video_dev/Kconfig"
@@ -84,8 +84,8 @@ comment "GPU (ARM Mali)"
 #isource "drivers/amlogic/ump/Kconfig"
 #isource "drivers/amlogic/mali/Kconfig"
 #comment "GPU (ARM Mali (external))"
-isource "../hardware/arm/gpu/ump/Kconfig"
-isource "../hardware/arm/gpu/mali/Kconfig"
+isource "drivers/amlogic/ump/Kconfig"
+isource "drivers/amlogic/mali/Kconfig"
 source "drivers/amlogic/ion_dev/Kconfig"
 
 #

