--- linux-amlogic-3.10-bringup/drivers/video/fbmem.c.orig	2014-04-30 17:01:54.274643025 +0800
+++ linux-amlogic-3.10-bringup/drivers/video/fbmem.c	2014-04-30 17:03:14.426644516 +0800
@@ -910,6 +910,13 @@ fb_pan_display(struct fb_info *info, str
 	return 0;
 }
 
+static int fb_cursor(struct fb_info *info, struct fb_cursor *var)
+{
+	if (info->fbops->fb_cursor)
+		return info->fbops->fb_cursor(info, var);
+	return -EINVAL;
+}
+
 static int fb_check_caps(struct fb_info *info, struct fb_var_screeninfo *var,
 			 u32 activate)
 {
@@ -1084,6 +1091,7 @@ static long do_fb_ioctl(struct fb_info *
 	struct fb_cmap cmap_from;
 	struct fb_cmap_user cmap;
 	struct fb_event event;
+	struct fb_cursor cursor;
 	void __user *argp = (void __user *)arg;
 	long ret = 0;
 
@@ -1145,7 +1153,16 @@ static long do_fb_ioctl(struct fb_info *
 			return -EFAULT;
 		break;
 	case FBIO_CURSOR:
-		ret = -EINVAL;
+		if (copy_from_user(&cursor, argp, sizeof(cursor)))
+			return -EFAULT;
+		if (!lock_fb_info(info))
+			return -ENODEV;
+		console_lock();
+		ret = fb_cursor(info, &cursor);
+		console_unlock();
+		unlock_fb_info(info);
+		if (ret == 0 && copy_to_user(argp, &cursor, sizeof(cursor)))
+			return -EFAULT;
 		break;
 	case FBIOGET_CON2FBMAP:
 		if (copy_from_user(&con2fb, argp, sizeof(con2fb)))
