diff -Nur -Nur qt-everywhere-opensource-src-4.8.2/mkspecs/qws/linux-amlogic8726M-arm-g++/qmake.conf qt-4.8.2/mkspecs/qws/linux-amlogic8726M-arm-g++/qmake.conf
--- qt-everywhere-opensource-src-4.8.2/mkspecs/qws/linux-amlogic8726M-arm-g++/qmake.conf	1970-01-01 08:00:00.000000000 +0800
+++ qt-4.8.2/mkspecs/qws/linux-amlogic8726M-arm-g++/qmake.conf	2012-06-12 17:04:31.576292568 +0800
@@ -0,0 +1,50 @@
+#
+# qmake configuration for building with arm-linux-g++
+#
+
+MAKEFILE_GENERATOR  = UNIX
+TARGET_PLATFORM     = unix
+TEMPLATE        = app
+CONFIG          += qt warn_on release incremental link_prl gdb_dwarf_index
+QT          += core gui
+QMAKE_INCREMENTAL_STYLE = sublib
+
+include(../common/linux.conf)
+include(../common/gcc-base-unix.conf)
+include(../common/g++-unix.conf)
+
+ROOTFS=/home/girish/amlogic/rootfs
+TOOLCHAIN=/home/girish/toolchains/arm-2010q1/bin
+
+# modifications to g++.conf
+# QMAKE_LFLAGS = -L$${ROOTFS}/lib -L$${ROOTFS}/usr/lib
+QMAKE_CXXFLAGS = -Os -pipe -Os  -mtune=cortex-a9 -march=armv7-a -mabi=aapcs-linux -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
+QMAKE_CFLAGS = -Os -pipe -Os  -mtune=cortex-a9 -march=armv7-a -mabi=aapcs-linux -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 
+
+# modifications to linux.conf
+QMAKE_STRIP =      /home/girish/toolchains/arm-2010q1/bin/arm-none-linux-gnueabi-strip
+QMAKE_RANLIB =     /home/girish/toolchains/arm-2010q1/bin/arm-none-linux-gnueabi-ranlib
+QMAKE_OBJCOPY =    /home/girish/toolchains/arm-2010q1/bin/arm-none-linux-gnueabi-objcopy
+QMAKE_AR =         /home/girish/toolchains/arm-2010q1/bin/arm-none-linux-gnueabi-ar cqs
+QMAKE_LINK_SHLIB = /home/girish/toolchains/arm-2010q1/bin/arm-none-linux-gnueabi-g++ --sysroot=$${ROOTFS} 
+QMAKE_LINK =       /home/girish/toolchains/arm-2010q1/bin/arm-none-linux-gnueabi-g++ --sysroot=$${ROOTFS}
+QMAKE_CXX =        /home/girish/toolchains/arm-2010q1/bin/arm-none-linux-gnueabi-g++ --sysroot=$${ROOTFS}
+QMAKE_CC =         /home/girish/toolchains/arm-2010q1/bin/arm-none-linux-gnueabi-gcc --sysroot=$${ROOTFS}
+
+# QMAKE_INCDIR += $${ROOTFS}/usr/include
+# QMAKE_LIBDIR += $${ROOTFS}/usr/lib
+# QMAKE_INCDIR_EGL +=  $${ROOTFS}/usr/include
+# QMAKE_LIBDIR_EGL +=  $${ROOTFS}/usr/lib
+QMAKE_LIBS_EGL   =   -lEGL
+
+# QMAKE_INCDIR_OPENGL =  $$QMAKE_INCDIR_EGL
+# QMAKE_LIBDIR_OPENGL =  $$QMAKE_LIBDIR_EGL
+# QMAKE_LIBS_OPENGL   =  $$QMAKE_LIBS_EGL -lGLESv2 
+# QMAKE_LIBS          =  -ljpeg -lpng -lz
+# 
+# 
+# QMAKE_INCDIR_OPENGL_ES2 = $$QMAKE_INCDIR_EGL
+# QMAKE_LIBDIR_OPENGL_ES2 = $$QMAKE_LIBDIR_EGL
+
+
+load(qt_config)
diff -Nur -Nur qt-everywhere-opensource-src-4.8.2/mkspecs/qws/linux-amlogic8726M-arm-g++/qplatformdefs.h qt-4.8.2/mkspecs/qws/linux-amlogic8726M-arm-g++/qplatformdefs.h
--- qt-everywhere-opensource-src-4.8.2/mkspecs/qws/linux-amlogic8726M-arm-g++/qplatformdefs.h	1970-01-01 08:00:00.000000000 +0800
+++ qt-4.8.2/mkspecs/qws/linux-amlogic8726M-arm-g++/qplatformdefs.h	2012-06-12 17:04:31.576292568 +0800
@@ -0,0 +1,42 @@
+/****************************************************************************
+**
+** Copyright (C) 2011 Nokia Corporation and/or its subsidiary(-ies).
+** All rights reserved.
+** Contact: Nokia Corporation (qt-info@nokia.com)
+**
+** This file is part of the qmake spec of the Qt Toolkit.
+**
+** $QT_BEGIN_LICENSE:LGPL$
+** GNU Lesser General Public License Usage
+** This file may be used under the terms of the GNU Lesser General Public
+** License version 2.1 as published by the Free Software Foundation and
+** appearing in the file LICENSE.LGPL included in the packaging of this
+** file. Please review the following information to ensure the GNU Lesser
+** General Public License version 2.1 requirements will be met:
+** http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
+**
+** In addition, as a special exception, Nokia gives you certain additional
+** rights. These rights are described in the Nokia Qt LGPL Exception
+** version 1.1, included in the file LGPL_EXCEPTION.txt in this package.
+**
+** GNU General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU General
+** Public License version 3.0 as published by the Free Software Foundation
+** and appearing in the file LICENSE.GPL included in the packaging of this
+** file. Please review the following information to ensure the GNU General
+** Public License version 3.0 requirements will be met:
+** http://www.gnu.org/copyleft/gpl.html.
+**
+** Other Usage
+** Alternatively, this file may be used in accordance with the terms and
+** conditions contained in a signed written agreement between you and Nokia.
+**
+**
+**
+**
+**
+** $QT_END_LICENSE$
+**
+****************************************************************************/
+
+#include "../linux-g++/qplatformdefs.h"
diff -Nur -Nur qt-everywhere-opensource-src-4.8.2/mkspecs/qws/linux-amlogic8726M-arm-g++/README qt-4.8.2/mkspecs/qws/linux-amlogic8726M-arm-g++/README
--- qt-everywhere-opensource-src-4.8.2/mkspecs/qws/linux-amlogic8726M-arm-g++/README	1970-01-01 08:00:00.000000000 +0800
+++ qt-4.8.2/mkspecs/qws/linux-amlogic8726M-arm-g++/README	2012-06-12 17:04:31.576292568 +0800
@@ -0,0 +1,11 @@
+Assumption:
+Amlogic reference software is built on Host PC as conveyed in Amlogic's documentation.
+
+Amlogic 8726M:
+Qt version tested: Qt 4.8.0
+Ref Build System: arm-2010q1-202-arm-none-linux-gnueabi-i686-pc-linux-gnu.tar.bz2 crosstools from http://openlinux.amlogic.com/download/linux/ARM/gnutools/
+
+Configure Qt as:
+/path/to/qt-everywhere-opensource-src-4.8.0/configure -force-pkg-config -embedded arm -opensource -confirm-license -largefile -no-qt3support -depths 8,16,24,32 -release -shared -little-endian -qt-gif -no-libmng -system-zlib -system-libjpeg -system-libpng -qt-libtiff -xmlpatterns -exceptions -plugin-gfx-simplegl -declarative -webkit -script -scripttools -prefix /usr -hostprefix ${MY_BUILD_TREE}/staging/usr -fast -no-rpath -nomake examples -nomake demos -opengl es2 -no-openvg -qt-mouse-pc -qt-mouse-linuxtp -qt-mouse-linuxinput -qt-kbd-tty -qt-kbd-linuxinput -make tools
+make -j4
+make install
